index.d.ts

```diff
@@ -21,20 +21,29 @@ export interface Nodeback<E, R> {
   (err: E | null, value?: R): void
 }
 
-export interface ConcurrentFutureInstance<L, R> {
+declare const $T: unique symbol
+
+export interface Functor<A> {
+  [$T]: unknown
+  'fantasy-land/map'<B extends this[typeof $T]>(mapper: (value: A) => B): Functor<B>
+}
+
+type Unfunctor<F extends Functor<unknown>, B> = ReturnType<(F & { [$T]: B })['fantasy-land/map']>
```

Comment:
I don't think this is really `Unfunctor`; I think it's more like `Refunctor`, as it takes in `<SomeFunctor<A>, B>` and returns `SomeFunctor<B>`.

Also, it might be worthwhile to add
```ts
type Unfunctor<F extends Functor<unknown>> = F extends Functor<infer T> ? T : never;
```
and to use that in the first line of `map`; atm it's a pretty long, hard-to-read line.

Reply:
Good ideas. I wasn't too happy about the Unfunctor name either, but I find that thing hard to name.

Reply:
Maybe `Mapped`?

Reply:
> it might be worthwhile to add [an alias for inferring the inner value of a Functor]

Actually, it seems it's only ever going to be used in one place.